### PR TITLE
feat(validation): allow credit card numbers with spaces

### PR DESCRIPTION
**Pull request by jike212**
## Description
When using input masking libraries like [Cleave.js](https://github.com/nosir/cleave.js/) in conjunction with Fomantic form validation rules for payments, legitimate card numbers from Fomantic's test cards would not validate if it contained spaces.

## Testcase
JSFiddle showing the problem: https://jsfiddle.net/h8pe31qb/

JSFiddle showing the solution with the modified line in this PR (excuse the roughness, I simply pasted in the result of my `gulp build` with the change included: https://jsfiddle.net/og5a6x2w/

## Screenshots
Failure on valid credit card number:
image

After fix: No validation error
image

### CODE REVIEW

**Review by lubber-de — DISMISSED** [review]
LGTM
I created a jsfiddle which uses your code natively from a CDN

https://jsfiddle.net/lubber/5n9sfjyh/

File: src/definitions/behaviors/form.js [hunk 1]
```diff
@@ -1993,6 +1993,9 @@ $.fn.form.settings = {
 
       // allow dashes in card
       cardNumber = cardNumber.replace(/[\-]/g, '');
+      
+      // allow spaces in card
+      cardNumber = cardNumber.replace(/\s/g, '');
```
Comment (exoego) [inline src/definitions/behaviors/form.js:1998]: Nit) I think these regex can be consolidated.

```suggestion
      // allow dashes and spaces in card
      cardNumber = cardNumber.replace(/[\s\-]/g, '');
```

**Review by exoego — DISMISSED** [review]
LGTM
Left one nit comment for simpler code.

**Review by lubber-de — APPROVED** [review]
LGTM

**Comment by lubber-de** [comment]
@all-contributors please add @jike212 for code

**Pull request merged**